baseplate/config.py

```diff
@@ -335,30 +334,125 @@ def __init__(self):
         self.__dict__ = self
 
 
-def _parse_config_section(config, spec, root):
-    parsed = ConfigNamespace()
-    for key, parser_or_spec in spec.items():
-        assert "." not in key, "dots are not allowed in keys"
+class Parser(object):
+    """Base for config parsers."""
 
-        if root:
-            key_path = "%s.%s" % (root, key)
-        else:
-            key_path = key
-
-        if callable(parser_or_spec):
-            parser = parser_or_spec
-            raw_value = config.get(key_path, "")
-
-            try:
-                parsed[key] = parser(raw_value)
-            except Exception as e:
-                raise ConfigurationError(key, e)
-        elif isinstance(parser_or_spec, dict):
-            subspec = parser_or_spec
-            parsed[key] = _parse_config_section(config, subspec, root=key_path)
+    @classmethod
+    def from_spec(self, spec):
```

Comment:
Whoops, good point. I turned it into a `staticmethod` and removed the unused first parameter altogether.

As for why it's on the class at all: I think it adds a bit of semantic connection to put it there following the static factory method pattern. There are a [couple](https://github.com/reddit/baseplate/blob/master/baseplate/core.py#L82-L123) [other](https://github.com/reddit/baseplate/blob/master/baseplate/secrets/store.py#L77-L90) places in baseplate I've done the same. (I think I've reintroduced this to my vocabulary via Rust, so if we feel it's not very pythonic I can stop doing it.)